nntrainer/layers/pow_layer.cpp

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 SeungBaek Hong <sb92.hong@samsung.com>
+ *
+ * @file   pow_layer.cpp
+ * @date   20 Nov 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author SeungBaek Hong <sb92.hong@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is pow layer class (operation layer)
+ *
+ */
+
+#include "common_properties.h"
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <pow_layer.h>
+#include <util_func.h>
+
+#include <layer_context.h>
+
+namespace nntrainer {
+
+void PowLayer::finalize(InitLayerContext &context) {
+  context.setOutputDimensions({context.getInputDimensions()[0]});
+}
+
+void PowLayer::forwarding_operation(const Tensor &input, Tensor &hidden) {
+  float exp = std::get<props::Exponent>(pow_props).get();
+  input.pow(exp, hidden);
+}
```

Comment:
Quick question! this isn't really has to do with PR but:
Is there any difference with multiplying itself if the given exponent is set to 2? ( or inv_sqrt function <-> pow with -0.5 exponent )
I find almost every `pow` or `pow_i` function usage in the nntrainer use 2.0, 0.5, or -0.5 as an exponent, and this is handled with naive loop Tensor member function.